geode-dunit/src/main/java/org/apache/geode/test/junit/rules/ServerStarterRule.java

```diff
@@ -167,6 +171,23 @@ public ServerStarterRule withRegion(RegionShortcut type, String name) {
     return this;
   }
 
+  public Region createRegion(RegionShortcut type, String name,
```

Comment:
Could you add a javadoc explaining what the consumer parameter is for?
I think it allows you to modify the factory that will be used to create the region before create is called on that factory